### PR TITLE
[SPARK-42816][CONNECT] Support Max Message size up to 128MB

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?

This change lifts the default message size of 4MB to 128MB and makes it configurable. While 128MB is a "random number" it supports creating DataFrames from reasonably sized local data without failing.

### Why are the changes needed?
Usability

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manual

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 1]
```diff
@@ -47,6 +47,13 @@ object Connect {
       .bytesConf(ByteUnit.MiB)
       .createWithDefaultString("4m")
 
+  val CONNECT_GRPC_MAX_INBOUND_MESSAGE_SIZE =
+    ConfigBuilder("spark.connect.grpc.maxInboundMessageSize")
+      .doc("Set the default inbound message size for the GRPC requests.")
```
Comment (pan3793) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:52]: nit: "GRPC" -> "gRPC"

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 2]
```diff
@@ -47,6 +47,13 @@ object Connect {
       .bytesConf(ByteUnit.MiB)
       .createWithDefaultString("4m")
 
+  val CONNECT_GRPC_MAX_INBOUND_MESSAGE_SIZE =
+    ConfigBuilder("spark.connect.grpc.maxInboundMessageSize")
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:51]: If this is a user-facing conf, should probably document at https://github.com/apache/spark/blob/master/docs/configuration.md#spark-connect

**Review by HyukjinKwon — APPROVED** [review]
cc @hvanhovell @zhengruifeng @WeichenXu123

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:51]: +1 for documenting.

File: python/pyspark/sql/connect/client.py [hunk 3]
```diff
@@ -122,6 +122,8 @@ class ChannelBuilder:
     PARAM_TOKEN = "token"
     PARAM_USER_ID = "user_id"
     PARAM_USER_AGENT = "user_agent"
+    MAX_MESSAGE_LENGTH = 128 * 1024 * 1024
+    MAX_METADATA_LENGTH = 16 * 1024 * 1024
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/connect/client.py:126]: This seems to be unused in this PR. Do we need to add this in this PR?

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 2 above)
Comment (amaliujia) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:51]: +1

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 4]
```diff
@@ -47,6 +47,13 @@ object Connect {
...
+      .doc("Set the default inbound message size for the GRPC requests.")
+      .version("3.4.0")
+      .bytesConf(ByteUnit.MiB)
```
Comment (amaliujia) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:54]: Should be `ByteUnit.BYTE`?

File: python/pyspark/sql/connect/client.py (hunk 3 above)
Comment (grundprinzip) [reply python/pyspark/sql/connect/client.py:126]: No, this is a leftover. Removing.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 2 above)
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:51]: Done

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 4 above)
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:54]: Done

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 5]
```diff
@@ -47,6 +47,16 @@ object Connect {
       .bytesConf(ByteUnit.MiB)
       .createWithDefaultString("4m")
 
+  val CONNECT_GRPC_MAX_INBOUND_MESSAGE_SIZE =
+    ConfigBuilder("spark.connect.grpc.maxInboundMessageSize")
+      .doc("""
+          |Sets the maximum inbound message size for the gRPC requests.
+          |Requests with a larger payload will fail.
+          |"""")
```
Comment (MaxGekk) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:55]:
```suggestion
          |""")
```

**Review by MaxGekk — COMMENTED** [review]
Is there any test which confirms that new settings really work.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 6]
```diff
@@ -47,6 +47,16 @@ object Connect {
       .bytesConf(ByteUnit.MiB)
       .createWithDefaultString("4m")
 
+  val CONNECT_GRPC_MAX_INBOUND_MESSAGE_SIZE =
+    ConfigBuilder("spark.connect.grpc.maxInboundMessageSize")
+      .doc("""
+          |Sets the maximum inbound message size for the gRPC requests.
```
[the remark this answers is not in the document]
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:53]: the unit is in bytes

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 7]
```diff
@@ -47,6 +47,16 @@ object Connect {
...
+          |Requests with a larger payload will fail.
```
[the remark this answers is not in the document]
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:54]: Fixed.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 5 above)
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:55]: fixed

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (grundprinzip) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala]: updated the doc.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master/branch-3.4. Thank you, @grundprinzip and all!

**Pull request closed without merging**